File: ch04_thermalHydraulics/matlab/kfuel_plot.m
clear
close all

x = 0.1; % U10Zr

kU   = @(T)(21.73 + 1.591e-2*T + 5.907e-6*T^2);
kZr  = @(T)(8.853 + 7.082e-3*T + 2.533e-6*T^2 + 2.992e3/T);
kcZr = @(T,x)(-102.0 + 200.1*x - 109.2*x^2 +9.453e-3*T + ...
    3.459e-5*T^2 - 0.02093*x*T);
kUZr = @(T,x)((1 - sqrt(1-x))*kZr(T) + sqrt(1-x)*((1-x)*kU(T) + x*kcZr(T,x)));

kHT9 = 64.33; % [W/m/K]
kNa  = 25.81; % [W/m/K]

Ttest = linspace(400,1200);
ktest = zeros(size(Ttest));
for i = 1:length(Ttest)
    ktest(i) = kUZr(Ttest(i),x);
end

LW = 3;

figure
hold on
plot(Ttest,ktest,'LineWidth',LW);
plot([Ttest(1),Ttest(end)],[kHT9,kHT9],'--','LineWidth',LW)
plot([Ttest(1),Ttest(end)],[kNa,kNa],'-.','LineWidth',LW)
hold off
xlim([Ttest(1),Ttest(end)])
xlabel('T [K]')
ylabel('Thermal Conductivity [W/m/K]')
title('Thermal Conductivity of Reactor Materials')
legend({'k_{U10Zr}','k_{HT9}','k_{Na}'})
set(gca,'FontName','Times New Roman','FontSize',12);
print('../figs/kfuel_plot.eps','-depsc2');
% close(gcf)

File: ch04_thermalHydraulics/python/kfuel_plot.py
import numpy as np
import matplotlib.pyplot as plt

LW = 2
FN = 'Times New Roman'
FS = 12
plt.rc('lines', lw=LW)
plt.rc('mathtext', fontset='stix')
plt.rc('font', family=FN, size=FS)

kU   = lambda T : (21.73 + 1.591e-2*T + 5.907e-6*T**2);
kZr  = lambda T : (8.853 + 7.082e-3*T + 2.533e-6*T**2 + 2.992e3/T);
kcZr = lambda T, x : (-102.0 + 200.1*x - 109.2*x**2 +9.453e-3*T + 3.459e-5*T**2 - 
    0.02093*x*T);
kUZr = lambda T, x : ((1 - np.sqrt(1-x))*kZr(T) + np.sqrt(1-x)*((1-x)*kU(T) + 
    x*kcZr(T,x)));

kHT9 = 64.33 # [W/m/K]
kNa  = 25.81 # [W/m/K]

x = 0.1 # U10Zr

Ttest = np.linspace(400., 1200., 100)
ktest = kUZr(Ttest, x)

plt.figure()
plt.plot(Ttest, ktest)
plt.plot([Ttest[0], Ttest[-1]], [kHT9, kHT9], '--')
plt.plot([Ttest[0], Ttest[-1]], [kNa,  kNa],  '-.')
plt.xlim([Ttest[0], Ttest[-1]])
plt.xlabel('T [K]')
plt.ylabel('Thermal Conductivity [W/m/K]')
plt.title('Thermal Conductivity of Reactor Materials')
plt.legend(['$k_{U10Zr}$', '$k_{HT9}$', '$k_{Na}$'])
plt.tight_layout()
plt.savefig('../figs/kfuel_plot.pdf', bbox_inches='tight', pad_inches=0)
plt.close()
